utils/gatekeeper.py

```diff
@@ -0,0 +1,33 @@
+from flask import current_app, request
+from functools import wraps
+from typing import Set
+
+
+def allowed_params(params: Set[str]):
+    def decorator(endpoint):
+        @wraps(endpoint)
+        def verify_fields(*args, **kwargs):
+            if not (request.json.keys() <= params):
+                return invalid_field_error(params)
+            return endpoint(*args, **kwargs)
+
+        return verify_fields
+
+    return decorator
+
+
+def invalid_field_error(field_set):
+    if current_app.env == "development":
+        raise GatekeeperError(field_set)
+    else:
+        return {"message": "Invalid request field"}, 400
```

Comment:
Let's switch this around. That way we raise the useful error in dev and test.
```suggestion
    if current_app.env == "production":
        return {"message": "Invalid request field"}, 400
    else:
        raise GatekeeperError(field_set)
```

Also, I'm not fond of using the string `"production"` for equality matcher.

That could change or easily be misspelled. I would like to define some constants in the env file for use across the app instead.